src/tests/test_generic_server.py

```diff
@@ -0,0 +1,165 @@
+import unittest
+import socket
+import Trace
+import mock
+import e_errors
+import option
+import generic_client
+import generic_server
+import mock_csc
+import event_relay_messages
+import enstore_erc_functions
+import hostaddr
+import udp_client
+
+
+class TestServerError(unittest.TestCase):
+    def setUp(self):
+        self.err = generic_server.ServerError('test')
+
+    def test___repr__(self):
+        self.assertEqual(self.err.__repr__(), "ServerError")
+
+
+class TestGenericServerInterface(unittest.TestCase):
+    def setUp(self):
+        self.gsi = generic_server.GenericServerInterface()
+
+    def test___init__(self):
+        self.assertTrue(isinstance(self.gsi, option.Interface))
+
+    def test_valid_dictionaries(self):
+        self.assertEqual(
+            self.gsi.valid_dictionaries(),
+            (self.gsi.help_options,
+             self.gsi.trace_options))
+
+
+class Msg(object):
+    def __init__(self, **kwargs):
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+
+
+class TestGenericServer(unittest.TestCase):
+    def setUp(self):
+        self.sent_msg = mock.MagicMock()
+        udp_client.UDPClient.send = self.sent_msg
+        udp_client.UDPClient.send_no_wait = self.sent_msg
+        self.csc = mock_csc.csc()
+        self.svr = generic_server.GenericServer(self.csc, 'test_server')
+
+    def test_handle_er_msg(self):
+        with mock.patch('enstore_erc_functions.read_erc') as read_erc:
+            read_erc.return_value = Msg(
+                type=event_relay_messages.NEWCONFIGFILE)
+            with mock.patch('generic_server.GenericServer._reinit2') as _reinit2:
+                self.svr.handle_er_msg(1)
+                self.assertTrue(_reinit2.called)
+
+    def test__reinit2(self):
+        log_contents = 'Received notification of new configuration file.'
+        with mock.patch('Trace.log') as log:
+            with mock.patch('generic_server.GenericServer._reinit') as _reinit:
+                self.svr._reinit2()
+                log_str = str(log.mock_calls)
+                self.assertTrue(log_contents in log_str, log_str)
+                self.assertTrue(_reinit.called)
+
+    def test__reinit(self):
```

Comment:
I think the most important call here is self.csc.new_config_obj.new_config_msg(), can we add a check for that? Seems like it should be easy with MockCSC?

Reply:
added check